.github/workflows/tox-test.yml

```diff
@@ -24,14 +24,26 @@ jobs:
       - name: Checkout repository
         uses: actions/checkout@v3
 
-      - name: Install Python Dependencies
-        uses: HassanAbouelela/actions/setup-python@setup-python_v1.4.0
-        id: setup_python
+      - name: Setup poetry
+        id: poetry_setup
+        uses: ItsDrike/setup-poetry@v1
         with:
-          install_args: "--with workflow-tox --without lint"
-          python_version: "${{ matrix.python-version }}"
+          install-args: "--with workflow-tox --without lint"
+          python-version: "${{ matrix.python-version }}"
+
+      - name: Pytest Cache Caching
+        uses: actions/cache@v3
+        with:
+          path: ${{ github.workspace }}/.pytest_cache
+          key: "pytest-${{ runner.os }}-${{ steps.poetry_setup.outputs.python-version }}-\
+                ${{ hashFiles('./tests/**', './poetry.lock', './pyproject.toml') }}"
+          # Restore keys allows us to perform a cache restore even if the full cache key wasn't matched.
+          # That way we still end up saving new cache, but we can still make use of the cache from previous
+          # version.
+          restore-keys: "pytest-${{ runner.os }}-${{ steps.poetry_setup.outputs-python-version}}-"
```

Comment:
```suggestion
          restore-keys: "pytest-${{ runner.os }}-${{ steps.poetry_setup.outputs.python-version}}-"
```

Reply:
Not relevant, as pytest cache will get removed, but good catch